src/contexts/CartContext.jsx

```diff
@@ -0,0 +1,156 @@
+import {createContext, useContext, useReducer, useEffect} from 'react'
+import axios from 'axios'
+import { useNavigate } from 'react-router-dom'
+import { cartReducer } from '../reducer/cartReducer'
+import { useAuth } from './AuthContext'
+import Toast from '../components/Toast/Toast'
+
+const CartContext = createContext()
+
+const CartProvider = ({children}) => {
+    const navigate = useNavigate()
+    const {loggedIn} = useAuth()
+
+    const initialCartState = {
+        cart: [],
+        cartCount: 0,
+        cartTotalPrice: 0,
+        cartTotalDiscount: 0,
+        cartFinalAmout: 0
+    }
+
+    const userToken = localStorage.getItem("userToken")
+
+    const [cartState, cartDispatch] = useReducer(cartReducer, initialCartState)
```

Comment:
here not require to include explicitly  state in variables names beacuse community knows that in useState or useReducer first variable holds the state and seconda variable is the function to set the state